Task 2/Task 2.1.2/Task 2.1.2/Program.cs

```diff
@@ -0,0 +1,804 @@
+                   $"длина линии - {Math.Round(Length, 2, MidpointRounding.AwayFromZero)}";
+        }
+    }
+
+    class Triangle : FlatFigures
+    {
+        public Triangle (float pointX, float pointY, float sideA, float sideB, float sideC)
+        {
+            StartpointX = pointX;
+            StartpointY = pointY;
+            SideA = sideA;
+            SideB = sideB;
+            SideC = sideC;
+        }
+
+        private float _sideA;
+
+        public float SideA
+        {
+            get => _sideA;
+            set => _sideA = value;
+        }
+
+        private float _sideB;
+
+        public float SideB
+        {
+            get => _sideB;
+            set => _sideB = value;
+        }
+
+        private float _sideC;
+
+        public float SideC
+        {
+            get => _sideC;
+            set => _sideC = value;
+        }
+
+        public override double GetSquare
+        {
+            get
+            {
+                float p = (SideA + SideA + SideC) / 2;
+                return Math.Sqrt(p * (p - SideA) * (p - SideB) * (p - SideC));
+            }
+        }
+
+        public override double GetPerimeter
+        {
+            get
+            {
+                return SideA + SideB + SideC;
+            }
+        }
+
+        public override string ToString()
+        {
+            return $"Треугольник.\n" +
+                   $"Координаты вершины теругольника - (x:, y:) {StartpointX} {StartpointY},\n" +
+                   $"площадь треугольника - {Math.Round(GetSquare, 2, MidpointRounding.AwayFromZero)}\n" +
+                   $"периметр треугольника - {Math.Round(GetPerimeter, 2, MidpointRounding.AwayFromZero)}";
+        }
+    }
+
+    class Quadrate : FlatFigures
+    {
+        public Quadrate(float pointX, float pointY, float side)
+        {
+            StartpointX = pointX;
+            StartpointY = pointY;
+            Side = side;
+        }
+
+        private float _side;
+
+        public float Side
+        {
+            get => _side;
+            set => _side = value;
+        }
+
+        public override double GetSquare
+        {
+            get
+            {
+                return Side * Side;
+            }
+        }
+
+        public override double GetPerimeter
+        {
+            get
+            {
+                return Side * 4;
+            }
+        }
+
+        public override string ToString()
+        {
+            return $"Квадрат.\n" +
+                   $"Координаты вершин квадрата - (x:, y:) {StartpointX} {StartpointY}, {StartpointX + Side} {StartpointY}, {StartpointX + Side} {StartpointY + Side}, {StartpointX} {StartpointY + Side}\n" +
+                   $"площадь квадрата - {Math.Round(GetSquare, 2, MidpointRounding.AwayFromZero)}\n" +
+                   $"периметр квадрата - {Math.Round(GetPerimeter, 2, MidpointRounding.AwayFromZero)}";
+        }
+    }
+
+    class Rectangle : Quadrate
```

Comment:
Square is a special case of a rectangle, you could just make a square out of rectangle.